aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -426,8 +424,7 @@ Tensor randn_like(const Tensor& self, const TensorOptions& options) {
 
 namespace {
 template <typename scalar_t>
-void randperm_cpu(Tensor& result, int64_t n, THGenerator* generator) {
-  std::lock_guard<std::mutex> lock(generator->mutex);
```

Comment:
I just realized we removed the `lock_guard` here. Is that OK? We are still using the generator right?

Reply:
Yeah. I moved the lock to `randperm_out_cpu`. I decided to keep the `get_generator_or_default` and the `lock_guard` next to each other, in most cases, to emphasize that we don't forgot taking out the lock.